searx/engines/semantic_scholar.py

```diff
@@ -34,13 +42,32 @@ def request(query, params):
 def response(resp):
     res = loads(resp.text)
     results = []
+
     for result in res['results']:
-        results.append(
-            {
-                'url': result['primaryPaperLink']['url'],
-                'title': result['title']['text'],
-                'content': result['paperAbstractTruncated'],
-            }
-        )
+        item = {}
+        metadata = []
+
+        url = result.get('primaryPaperLink', {}).get('url')
+        if not url and result.get('links'):
+            url = result.get('links')[0]
+        if not url:
+            continue
```

Comment:
Some paper have only link to the PDF instead of a description.
Example: 
* search for "Linux"
* The following papers are ignored:
  * "Docker: lightweight Linux containers for consistent development and deployment" is ignored 
  * kvm : the Linux Virtual Machine Monitor

To display them in the results, replace `continue` by:
```python
            alternatePaperLinks = result.get('alternatePaperLinks', [])
            if alternatePaperLinks:
                url = alternatePaperLinks[0]['url']
            else:
                continue
```

---

Extract of the JSON output for the "alternatePaperLinks" field for one paper:

```json
    'alternatePaperLinks': [{
        'url': 'https://www.seltzer.com/margo/teaching/CS508.19/papers/merkel14.pdf',
        'linkType': 'anansi'
    }],
```

I don't know what `anansi` means

Reply:
Good point / thanks! .. I changed the commit by:

```diff 
@@ -14,7 +14,10 @@ about = {
     "require_api_key": False,
     "results": 'JSON',
 }
+
+paging = True
 search_url = 'https://www.semanticscholar.org/api/1/search'
+paper_url = 'https://www.semanticscholar.org/paper'
 

def request(query, params):
@@ -51,7 +54,12 @@ def response(resp):
         if not url and result.get('links'):
             url = result.get('links')[0]
         if not url:
-            continue
+            alternatePaperLinks = result.get('alternatePaperLinks')
+            if alternatePaperLinks:
+                url = alternatePaperLinks[0].get('url')
+        if not url:
+            url = paper_url + '/%s' % result['id']
+
         item['url'] = url
 
         item['title'] = result['title']['text']
```

As you can see I forgot paging .. and when there is no url available, we now link to the paper page at https://www.semanticscholar.org/paper